scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The log-PDF evaluated at the argument `x`.
+
+        See Also
+        --------
+        ContinuousDistribution.pdf
+        ContinuousDistribution.logcdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`.
+        By definition of the support, the log-PDF evaluates to its minimum value
+        of :math:`-\infty` (i.e. :math:`\log(0)`) outside the support; i.e. for
+        :math:`x < l` or :math:`x > r`. The maximum of the log-PDF may be less
+        than or greater than :math:`\log(1) = 0` because the maximum of the PDF
+        can be any positive real.
+
+        For distributions with infinite support, it is common for
+        `ContinuousDistribution.pdf` to return a value of `0` when the argument
+        is theoretically within the support; this can occur because the true value
+        of the PDF is too small to be represented by the chosen dtype. The log
+        of the PDF, however, will often be numerically nonzero over a much larger
+        domain. Consequently, it may be preferred to work with the logarithms of
+        probabilities and probability densities to avoid underflow.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1.0, b=1.0)
+
+        Evaluate the log-PDF at the desired argument:
+
+        >>> X.logpdf(0.5)
+        -0.6931471805599453
+        >>> np.allclose(X.logpdf(0.5), np.log(X.pdf(0.5)))
+        True
+
+        """
+        return self._logpdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _logpdf_dispatch(self, x, *, method=None, **kwargs):
+        if self._overrides('_logpdf_formula'):
+            method = self._logpdf_formula
+        elif self.tol is _null:  # ensure that developers override _logpdf
+            method = self._logpdf_logexp
+        return method
+
+    def _logpdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _logpdf_logexp(self, x, **kwargs):
+        return np.log(self._pdf_dispatch(x, **kwargs))
+
+    @_set_invalid_nan
+    def pdf(self, x, *, method=None):
+        r"""Probability density function
+
+        The probability density function :math:`f(x)` is the probability *per
+        unit interval* of :math:`x` that the random variable takes on the value
+        :math:`x`. Mathematically, it is defined as the derivative of the
+        cumulative distribution function `F(x)`:
+
+        .. math::
+
+            f(x) = \frac{d}{dx} F(x)
+
+        ``pdf`` accepts `x` for :math:`x`.
+
+        Parameters
+        ----------
+        x : Array
+            The value of the random variable for which the PDF is to be
```

Comment:
drop "of the random variable"?

Reply:
Rationale?

Reply:
+1 to Christiph's sugggestion

Reply:
Rationale?

Reply:
I will change to be more similar to what I wrote for CDF, etc.

Reply:
The rationale is to avoid the confusion you described: X is the random variable, x is just a value in the support of X (or even any real number x where pdf(x) = 0 outside the support)

Reply:
K. Done.